dbms/src/Common/getNumberOfLogicalCPUCores.h

```diff
@@ -14,6 +14,13 @@
 
 #pragma once
 
-/// Get number of CPU cores without hyper-threading.
-/// Note: do not support environment under resource isolation mechanism like Docker, CGroup.
-unsigned getNumberOfPhysicalCPUCores();
+#include <thread>
+
+extern UInt16 max_logical_cpu_cores;
```

Comment:
I think it is a bad idea to define a global var without any namespace. And it is only defined in `Server.cpp`, I wonder it will cause compile error on ut binary.

better define it as a static var under `namespace details::cpu_info { ... }`